qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -91,20 +109,23 @@ def _compute_gradients(
             List of pairs consisting of gradient and excitation operator.
         """
         res = []
-        # compute gradients for all excitation in operator pool
         for exc in self._excitation_pool:
             # add next excitation to ansatz
             self._ansatz.operators = self._excitation_list + [exc]
-            # set the current ansatz
-            vqe.ansatz = self._ansatz
-            # evaluate energies
-            parameter_sets = theta + [-self._delta] + theta + [self._delta]
-            energy_evaluation = vqe.get_energy_evaluation(self._main_operator)
-            energy_results = energy_evaluation(np.asarray(parameter_sets))
+            if self.gradient.grad_method.analytic:
```

Comment:
As suggested by @Cryoris in the Qiskit Demo Day where this work was presented, we should dig a little deeper into why this is necessary. Maybe this requires a fix on the Terra level instead of this conditional check, here.